ARCs/arc-0018.md

```diff
@@ -0,0 +1,348 @@
+---
+arc: 18
+title: Royalty Enforcement Specification 
+description: An ARC to specify the methods and mechanisms to enforce Royalty payments as part of ASA transfers 
+author: Cosimo Bassi (@cusma), Ben Guidarelli (@barnjamin), Steve Ferrigno (@nullun) 
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): Interface
+created: 2022-02-16
+requires: ARC-4
+---
+
+## Abstract
+
+A specification to describe a mechanism to enforce [Royalty Payments](https://en.wikipedia.org/wiki/Royalty_payment) to a Royalty Collector given a policy describing the royalty shares, both on primary and secondary sales.
+
+## Motivation
+
+This ARC is defined to provide a consistent set of asset configurations and ABI methods that, together, enable a royalty payment to a Royalty Collector.
+
+An example may include some music rights where the label, the artist, and any investors have some assigned royalty percentage that should be enforced on transfer. During the sale transaction, the appropriate royalty payments should be included or the transaction must be rejected.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+
+[Royalty Policy](#royalty-policy) - The data structure that defines how royalties are dispersed
+
+[Royalty Enforcer](#royalty-enforcer) - The contract that enforces the royalty payments given the Royalty Policy and performs transfers of the assets.
+
+[Royalty Collector](#royalty-collector) - The account that collect the royalty and provides the royalty distribution logic. It could be set to a Single Signature, a Multi Signature, a Smart Signature or even another Smart Contract.
```

Comment:
> One thing that we may want to require however is that it necessarily opted in to the allowed assets for payments.
> Also, we may want to specify what happens if it does not:
> does it mean the asset is no more transferrable? or that royalties are not collected?

[From my comment here](https://github.com/algorandfoundation/ARCs/pull/70#discussion_r838331343): I want to avoid (and be able to handle somehow) the "lock-in effect" by which people are no longer able to transfer and trade their ASA not because they refuse to comply with the `Royalty Policy` but because that Policy is broken by the creator ex-post.

If `Royalty Collector` opts-out all the `Share ASA` I would expect that the `Royalty ASA` is now “royalty free” (since the `Royalty Collector` doesn't want to receive Share ASA at all).